### PR TITLE
(on getgov-za) Ticket #1947: Contact info on the domain information inline

### LINKED ISSUE #1947: Django Admin: For domains, add contact info inline and remove unnecessary multi-selects
### Issue description

[This PR](https://github.com/cisagov/manage.get.gov/pull/1927) (based on #1852 and #1836) added contact info inline and removed a few multiselect fields (made them permanently readonly) on the Domain Request page. For consistency, we should also implement these changes to the Domain model in Django Admin. More specifically, we should target the "Domain Information" inline and have it exactly match the style we defined in the Domain Request model.

### Acceptance criteria
On the django admin Domain page the following should be inplace:
- [ ] Inline contact information is present for Creator, Submitter, Authorizing Official, and other employees
- [ ] Other employees is readonly for analysts AND superusers
- [ ] The fieldset lines creator, submitter, authorizing_official, and other_employees all behave like the lines do in the DomainInformation model on Django Admin and mimic the domain request layout for these. 
- [ ] Unit tests

### Additional context
See this [thread](https://github.com/cisagov/manage.get.gov/pull/1927#discussion_r1540171150) for additional information.

image
image

### Links to other issues

🔄#1852, #1836

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1947

## Changes

- Moves readonly fields to DomainInformationInline
- Creates a "stacked.html" override for custom inline stylings
- Modifies the detail_table_fieldset file to instead track original through a shared variable, as to keep things consistent
- Adds unit tests
- Associates the inline override

## Context for reviewers
This PR adds contact information to the Domain Information inline.

## Setup
As an analyst, navigate to any domain. Scroll down to the DomainInformation section, all information there should be accurate and consistent with the format/functionality of DomainRequest

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/templates/django/admin/domain_information_change_form.html [hunk 1]
```diff
@@ -10,6 +10,6 @@
 
         Use detail_table_fieldset as an example, or just extend it.
         {% endcomment %}
-        {% include "django/admin/includes/detail_table_fieldset.html" %}
+        {% include "django/admin/includes/detail_table_fieldset.html" with original_object=original %}
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/django/admin/domain_information_change_form.html:13]: I don't get it

**Comment by rachidatecs** [comment]
Non-blocking - easy fix. In Firefox:

Screenshot 2024-04-17 at 2 58 55 PM

Fix:

Screenshot 2024-04-17 at 2 58 48 PM

File: src/registrar/admin.py [hunk 2]
```diff
@@ -1490,6 +1491,7 @@ class DomainInformationInline(admin.StackedInline):
             del fieldsets[index]
             break
 
+    readonly_fields = DomainInformationAdmin.readonly_fields
```
[the remark this answers is not in the document]
Comment (abroddrick) [reply src/registrar/admin.py:1494]: Can you elaborate more on what needs to be cleaned up? Not sure I have the full context here on what's lingering (save me from searching through the codebase).

File: src/registrar/templates/django/admin/domain_information_change_form.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/django/admin/domain_information_change_form.html:13]: Got it. Worth a comment in code.

**Review by rachidatecs — APPROVED** [review]
Minor comment suggestion

**Pull request merged**